argus/LoginProviders/String+Extension.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  String+Extension.swift
+//  argus
+//
+//  Created by Amit Kumar Swami on 08/09/17.
+//  Copyright © 2017 moldedbits. All rights reserved.
+//
+
+import Foundation
+
+extension String {
+    func isValidEmail() -> Bool {
+        // here, `try!` will always succeed because the pattern is valid
+        let regex = try! NSRegularExpression(pattern: "^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$", options: .caseInsensitive)
+        return regex.firstMatch(in: self, options: [], range: NSRange(location: 0, length: characters.count)) != nil
+    }
+    
+    func containsSpecialCharacter() -> Bool {
+        let characterset = CharacterSet(charactersIn: "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
```

Comment:
Can we not use CharacterSet.alphanumerics?

Reply:
No, this is something else.